### DISCUSSION TITLE
Short cannon-like burst animation on button click

### DISCUSSION THREAD

**Discussion by stylessard** [matteobruni/tsparticles#2940] [Q&A] [answered]
Hi! I'm trying to mimic the popular confetti cannon animation, but instead of emitting endlessly, I'd like a short burst instead and that's it. After finding a similar question in the discussion boards, I tried tinkering with the life.duration and life.count but haven't had any success. Any idea how I could achieve such an animation?

**Comment by matteobruni** [comment, marked as answer]
Hi @stylessard,

I've created a sample here: https://codepen.io/matteobruni/pen/wvPvOVg

It uses the `onClick` event with `emitters` mode

``` json
{
  "interactivity":{
    "events":{
      "onClick":{
        "enable":true,
        "mode":"emitter"
      }
    },
    "modes":{
      "emitters":{
        /* this accepts the emitters options */
      }
    }
  }
}
```

This is the basic configuration for spawning emitting particles on click, if you want a single explosion use `life.count: 1` in the emitters options.

**Reply by stylessard** [reply]
Perfect, that's exactly it! Just one more thing - going off the Codepen, would I be able to fire the confetti only when the text is clicked rather than anywhere on the canvas? You mentioned using `container.plugins.get("emitters").array[0].play()`; in a past issue from 2020, but I haven't been able to get that working.

**Reply by matteobruni** [reply]
Yes, I think that API is broken, but I was testing the new ones and they have some bugs too. I'm working on it right now. I'll give you an update ASAP.

**Reply by matteobruni** [reply]
Released `1.39.1` with the fix, checkout this pen here: https://codepen.io/matteobruni/pen/VwrYjxR

It uses the `addEmitter` method of the container, is available only if the emitter plugin is needed. The `emitters: []` is required to load the plugin, but since it's empty it does nothing to the configuration.

**Reply by stylessard** [reply]
Incredible! Thanks so much, Matteo!
